document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoQueryExecutor.java

```diff
@@ -79,6 +80,15 @@
     return iterable.cursor();
   }
 
+  private static Collection<BasicDBObject> reApplyFilterAfterUnwind(Query query) {
```

Comment:
This may clearly affect performance.
Can't filter be just done after unwind?

Reply:
The filter step before unwind is necessary to reduce the amount of documents unwind is applied to. 
Post filter is not affecting performance but yeah the forced filtering can reduce flexibility of this interface. 
As discussed will introduce a filter inside the UnnestExpression for flexible filtering post each unwind step.